python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -33,9 +33,46 @@
 logger = lsst.log.Log.getLogger("meas.deblender.deblend")
 
 
+def hasEdgeFlux(source, edgeDistance=1):
+    """hasEdgeFlux
+
+    Determine whether or not a source has flux within `edgeDistance`
+    of the edge.
+
+    Parameters
+    ----------
+    source : `scarlet.Component`
+        The source to check for edge flux
+    edgeDistance : int
+        The distance from the edge of the image to consider
+        a source an edge source. For example if `edgeDistance=3`
+        then any source within 3 pixels of the edge will be
+        considered to have edge flux
+
+    Returns
+    -------
+    isEdge: `bool`
+        Whether or not the source has flux on the edge.
+    """
+    assert edgeDistance > 0
+
+    # Use the first band that has a non-zero SED
+    band = np.min(np.where(source.sed > 0)[0])
+    model = source.get_model()[band]
+    for edge in range(edgeDistance):
+        if (
+            np.any(model[edge-1] > 0) or
```

Comment:
are scarlet models somehow 1 based indexing? if edge in the loop is zero, 0-1 is -1 and so this will test the last pixel, then the 0th etc. Is this what you were intending?

Reply:
Note [the assert](https://github.com/lsst/meas_extensions_scarlet/blob/tickets/DM-23364/python/lsst/meas/extensions/scarlet/source.py#L57). So the minimum value of `edgeDistance` is 1, which describes the edge.

I updated the docstring to make this clear.